test-e2e.sh

```diff
@@ -81,6 +81,22 @@ kubectl version
 # Don't build/test the prow image as it requires Docker
 EXCLUDED_TARGETS="-//images/gcloud-bazel:gcloud_install -//images/gcloud-bazel:gcloud_push"
 
+# Create a unique namespace for this job using the repo name and the build id
+export E2E_NAMESPACE="${REPO_OWNER:-default-owner}-${REPO_NAME:-default-name}-${BUILD_ID:-0}"
```

Comment:
I'd suggest waiting until we validate that `bazel build` and `bazel test` are successful before creating any kubernetes resources.